.github/workflows/benchmark.yml

```
@@ -0,0 +1,70 @@
+# This is a basic workflow to help you get started with Actions
+
+name: benchmark-check
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push or pull request events but only for the master branch
+  pull_request:
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      # This is too complicated to get working for now.
+      #
+      # # fetch the target branch from origin too (just the head)
+      # # and check it out into a local branch
+      # - name: Checkout PR and target branch
+      #   run: |
+      #     git fetch --no-tags --depth=1 origin ${{ github.base_ref }}
+      #     git fetch --no-tags --depth=1 origin ${{ github.head_ref }}
+
+      - name: Setup asv
+        run: |
+          pip install asv
+          cd benchmarks
+          asv machine --yes --machine gh-actions
+
+      - name: Run benchmarks on source and target
+        run: |
+          cd benchmarks
+          export BASE=origin/${{ github.base_ref }}
+          export HEAD=origin/${{ github.head_ref }}
+          # run asv benchmarks on the target
+          # -s 1 ensures only one "step" (the tip of the branch) is run
+          asv run $BASE -s 1
+          # run asv on the head of this PR
+          asv run $HEAD -s 1
```

```suggestion
          asv continuous $BASE $HEAD
```

This is designed specifically to error if regressions are found, so that you wouldn't need the `Fail if performance degraded` step.

Obviously some re-jigging would be needed so that we could still get `compare.txt`, or even to just have the raw `asv compare` output in the CI log - either seem valuable. But it would overall be simpler, and use ASV's intended API rather than parsing their text output.

---

how did I miss that...!!